### ISSUE TITLE
Using razor engine dynamically

### ISSUE THREAD

**Issue by TehWardy** [Antaris/RazorEngine#358]
Hey guys,

I'm trying to build an odata endpoint behind which I am sitting the razor engine to allow me to template lots of different things for different scenarios across my organisation. 

as part of the endpoint I have defined (on top of the usual CRUD functionality for maintaining template info) a render method. 

Here's what I'm trying to do ...

client request: 
```
POST "~/Template(1)/Render()"
model data { ... }
```

on the server:
```
return service.Render(int templateId, string modelData) {
      var template = db.Get<Template>(templateId);
      var model = ????;
      return RazorEngine.Engine.IsolatedRazor.RunCompile(
              template.Content, 
              template.Name, 
              null, 
              model
      );
}
```
In order to ensure that the model data I have been given conforms to that which is needed by my template I want to about ask the template what type of model it wants ...
In an ideal world I want to do is something like ...
```
template.Content = "@model Foo ..."
var modelType = RazorEngine.Engine.IsolatedRazor.GetModelTypeFrom(template.Content);
var model = ParseModelAs(modelType, modelData);
return RazorEngine.Engine.IsolatedRazor.RunCompile(
              template.Content, 
              template.Name, 
              null, 
              model
      );
```
Is this possible as I can't seem to find anything for this?

**Comment by matthid** [comment]
No this is not possible as templates can work without any model type as well. As a heuristic you can try to parse and resolve the type in the `@model` or `@inherits` line yourself (which don't need to exist at all and can even be overwritten by using a custom template-base-class).

**Comment by TehWardy** [comment]
Yes that's what i'm talking about ... if the model directive is in the template I should be able to detect it and then ensure the data I have can be cast to that type of data, but also knowing the type will help with deserialisation of my odata requested model json string

**Comment by matthid** [comment]
With the implementation on `GetModelTypeFrom` you are on your own. But feel free to contribute back with a PR :)
